packages/headless/src/features/search-parameters/search-parameter-serializer.test.ts

```diff
@@ -17,497 +22,565 @@ const someSpecialCharactersThatNeedsEncoding = [
-    });
+      it('does not include invalid keys', () => {
+        const invalidKeys = {invalid: 'b'};
+        const result = serialize({q: 'a', ...invalidKeys});
 
-    it('serializes the #fExcluded parameter correctly', () => {
-      const fExcluded = {author: ['a', 'b'], filetype: ['c', 'd']};
-      const result = serialize({fExcluded});
-      expect(result).toEqual('fExcluded-author=a,b&fExcluded-filetype=c,d');
-    });
+        expect(result).toBe('q=a');
+      });
 
-    it('serializes special characters in the #f parameter correctly', () => {
-      someSpecialCharactersThatNeedsEncoding.forEach((specialChar) => {
-        const f = {author: ['a', specialChar]};
+      it('serializes the #f parameter correctly', () => {
+        const f = {author: ['a', 'b'], filetype: ['c', 'd']};
         const result = serialize({f});
-        expect(result).toEqual(`f-author=a,${encodeURIComponent(specialChar)}`);
+        expect(result).toEqual('f-author=a,b&f-filetype=c,d');
       });
-    });
-
-    it('when the #f parameter contains invalid values, it does not include it', () => {
-      const f = {author: [1]} as never;
-      const result = serialize({f});
-      expect(result).toEqual('');
-    });
 
-    it('serializes the #nf parameter correctly', () => {
-      const nf = {
-        size: [
-          buildNumericRange({start: 0, end: 10}),
-          buildNumericRange({start: 10, end: 20}),
-        ],
-        amount: [buildNumericRange({start: 100, end: 200})],
-      };
+      it('serializes the #fExcluded parameter correctly', () => {
+        const fExcluded = {author: ['a', 'b'], filetype: ['c', 'd']};
+        const result = serialize({fExcluded});
+        expect(result).toEqual('fExcluded-author=a,b&fExcluded-filetype=c,d');
+      });
 
-      const result = serialize({nf});
-      expect(result).toEqual('nf-size=0..10,10..20&nf-amount=100..200');
-    });
+      it('serializes special characters in the #f parameter correctly', () => {
+        someSpecialCharactersThatNeedsEncoding.forEach((specialChar) => {
+          const f = {author: ['a', specialChar]};
+          const result = serialize({f});
+          expect(result).toEqual(
+            `f-author=a,${encodeURIComponent(specialChar)}`
+          );
+        });
+      });
 
-    describe('when the #nf parameter contains facetIds with invalid values', () => {
-      it('if a range does not have a #start key, it does not include it', () => {
-        const nf = {size: [{}]} as never;
-        const result = serialize({nf});
+      it('when the #f parameter contains invalid values, it does not include it', () => {
+        const f = {author: [1]} as never;
+        const result = serialize({f});
         expect(result).toEqual('');
       });
 
-      it('if a range does not have an #end key, it does not include it', () => {
-        const nf = {size: [{start: 10}]} as never;
+      it('serializes the #nf parameter correctly', () => {
```

Comment:
Should we also test that end inclusive ranges are serialized with the proper separator?